modes/settings_menues.h

```diff
@@ -73,43 +73,62 @@ class ShowLengthStyle {
   int len_;
 };
 
+template<class T>
+class CustomShowColorSingleBladeTemplate : public ShowColorSingleBladeTemplate<T, T> {
+public:
+  void SetStyle(BladeBase *blade, const char* str) {
+    blade->SetStyle(new Style<T>());
+  }
+
 template<class SPEC>
-struct ChangeBladeLengthBlade1 : public SPEC::MenuBase {
+struct ChangeBladeLengthMode : public SPEC::MenuBase {
   virtual int blade() { return menu_current_blade; }
   void mode_activate(bool onreturn) override {
     int len = prop_GetBladeLength(blade());
     int maxlen = prop_GetMaxBladeLength(blade());
-    saved_len_ = len;
     if (len == -1) len = maxlen;
-    this->pos_ = len;
+    saved_len_ = len;
+    pos_ = saved_len_;
     prop_SetBladeLength(blade(), maxlen);
     showlen_.Start(blade());
     SPEC::MenuBase::mode_activate(onreturn);
+    if (maxlen == 0) exit();
```

Comment:
If the user has manually set the blade length to 0 in the config, just exit as no length adjustment is possible.

Reply:
As per your response above, this should never really be because it wouldn't make sense. However, it could happen since it's not guarded in the BladeConfig, and/or it could be set intentionally in a no-blade scenario.

Reply:
Got it. Maybe we should throw an error if user config tries to set 0 length.